Content.Server/GameObjects/Components/Chemistry/SolutionComponent.cs

```diff
@@ -54,6 +135,85 @@ public void InitializeFromPrototype()
             _reactions = _prototypeManager.EnumeratePrototypes<ReactionPrototype>();
         }
 
+        /// <inheritdoc />
+        protected override void Shutdown()
+        {
+            base.Shutdown();
+
+            _containedSolution.RemoveAllSolution();
+            _containedSolution = new Solution();
+        }
+
+        public void RemoveAllSolution()
+        {
+            _containedSolution.RemoveAllSolution();
+            OnSolutionChanged();
+        }
+
+        public bool TryRemoveReagent(string reagentId, int quantity)
+        {
+            if (!ContainsReagent(reagentId, out var currentQuantity)) return false;
+
+            _containedSolution.RemoveReagent(reagentId, quantity);
+            OnSolutionChanged();
+            return true;
+        }
+
+        /// <summary>
+        /// Attempt to remove the specified quantity from this solution
+        /// </summary>
+        /// <param name="quantity">Quantity of this solution to remove</param>
+        /// <returns>Whether or not the solution was successfully removed</returns>
+        public bool TryRemoveSolution(int quantity)
+        {
+            if (CurrentVolume == 0)
+                return false;
+
+            _containedSolution.RemoveSolution(quantity);
+            OnSolutionChanged();
+            return true;
+        }
+
+        public Solution SplitSolution(int quantity)
+        {
+            var solutionSplit = _containedSolution.SplitSolution(quantity);
+            OnSolutionChanged();
+            return solutionSplit;
+        }
+
+        protected void RecalculateColor()
+        {
+            if(_containedSolution.TotalVolume == 0)
+                SubstanceColor = Color.White;
+
+            Color mixColor = default;
+            float runningTotalQuantity = 0;
+
+            foreach (var reagent in _containedSolution)
+            {
+                runningTotalQuantity += reagent.Quantity;
+
+                if(!_prototypeManager.TryIndex(reagent.ReagentId, out ReagentPrototype proto))
+                    continue;
+
+                if (mixColor == default)
+                    mixColor = proto.SubstanceColor;
+
+                mixColor = BlendRGB(mixColor, proto.SubstanceColor, reagent.Quantity / runningTotalQuantity);
```

Comment:
```suggestion
                mixColor = BlendRGB(mixColor, proto.SubstanceColor, reagent.Quantity / (float)runningTotalQuantity);
```
The last parameter is calculated with integer division which causes huge banding in the final color.